tests/test_network.py

```diff
@@ -124,6 +128,102 @@ def test_add_objective(factory, client):
+    ({'foo': 'bar'}, {'foo': 'bar'}, {'foo': {"bar": "foo"}}),
+    (None, {}, {too_long: "bar"}),
+    ({}, {}, {"foo": too_long}),
+])
+def test_metadata(factory, client, asset_name, metadata, metadata_output, metadata_error):
+    method_create = getattr(factory, f"create_{asset_name}")
+    method_add = getattr(client, f"add_{asset_name}")
+    method_sdk_add = getattr(client._client, f"add_{asset_name}")
+
+    # add an asset with metadata
+    spec = method_create(metadata=metadata)
+    asset = method_add(spec)
+
+    assert asset.metadata == metadata_output
+
+    spec = method_create()
+    spec_dict = spec.dict()
+    spec_dict['metadata'] = 'foo'
+
+    with pytest.raises(substra.exceptions.InvalidRequest):
+        method_sdk_add(spec_dict)
+
+    spec = method_create(metadata=metadata_error)
+    with pytest.raises(substra.exceptions.InvalidRequest):
+        method_add(spec)
+
+
+@pytest.mark.parametrize('asset_name, algo_type', [
+    ('traintuple', 'algo'),
+    ('composite_traintuple', 'composite_algo'),
+])
+@pytest.mark.parametrize('metadata,metadata_output,metadata_error', [
+    ({'foo': 'bar'}, {'foo': 'bar'}, {'foo': {"bar": "foo"}}),
+    (None, {}, {too_long: "bar"}),
+    ({}, {}, {"foo": too_long}),
+])
+def test_metadata_traintuple(factory, client, asset_name, algo_type, default_dataset,
+                             metadata, metadata_output, metadata_error):
+    method_create = getattr(factory, f"create_{asset_name}")
+    method_add = getattr(client, f"add_{asset_name}")
+
+    algo_create = getattr(factory, f"create_{algo_type}")
+    algo_add = getattr(client, f"add_{algo_type}")
+
+    algo_spec = algo_create()
+    algo = algo_add(algo_spec)
+    spec = method_create(
+        algo=algo,
+        dataset=default_dataset,
+        data_samples=default_dataset.train_data_sample_keys,
+        metadata=metadata
+    )
+    asset = method_add(spec)
+    assert asset.metadata == metadata_output
+
+    spec = method_create(
+        algo=algo,
+        dataset=default_dataset,
+        data_samples=default_dataset.train_data_sample_keys,
+        metadata=metadata_error
+    )
+    with pytest.raises(substra.exceptions.InvalidRequest):
+        method_add(spec)
+
+
+def test_metadata_aggregatetuple(factory, client, default_dataset):
+    composite_algo_spec = factory.create_composite_algo()
+    composite_algo = client.add_composite_algo(composite_algo_spec)
+    composite_traintuple_spec = factory.create_composite_traintuple(
+        algo=composite_algo,
+        dataset=default_dataset,
+        data_samples=default_dataset.train_data_sample_keys,
+    )
+    composite_traintuple = client.add_composite_traintuple(composite_traintuple_spec)
+
+    aggregate_algo_spec = factory.create_aggregate_algo()
+    aggregate_algo = client.add_aggregate_algo(aggregate_algo_spec)
+    spec = factory.create_aggregatetuple(
+        algo=aggregate_algo,
+        worker='MyOrg1MSP',
```

Comment:
The only hardcoded values in tests should be the values you're actually testing. In this case, The worker is `MyOrg1MSP` because it's the value of client.node_id, so you should use `client.node_id` instead.